framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -418,19 +430,41 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
-            if (postcondAnno == null) {
-                continue;
-            }
-            for (String expr : expressions) {
-                result.add(new Postcondition(expr, postcondAnno, anno));
+            Annotation p = methodElement.getAnnotation(EnsuresNonNull.List.class);
+            if (p == null) p = methodElement.getAnnotation(EnsuresKeyFor.List.class);
```

Comment:
Hard-coded logic is a code smell.  You should avoid that, because it is a bad practice.

As implemented, it is impossible for a type system designer to create their own wrapper annotations, without modifying the code of the Checker Framework and re-compiling it!  We do not want that.  I would expect that you can test for the existence of a meta-annotation, such as whatever you will need to define to replace the uses of `@PostconditionAnnotation`.

Reply:
You could also keep the test you had before (about the type of the value field).  The criticism of your previous approach was primarily about the fact that it used exceptions as control flow (and without any documentation of what it was doing).

Reply:
I have fixed this